347/step2_3_quickselect.py

```diff
@@ -0,0 +1,47 @@
+# 解法3: Quickselect（平均 O(n)）
+# 配列を（頻度で）k番目以降が右側に来るよう分割し、右側 k 個を返す
+# 平均的には全部を順番に並べる必要はないので早い
+# 時間計算量: 平均 O(m) / 空間: O(m)（Counter分）
+from collections import Counter
+from typing import List
+import random
+
+
+class Solution:
+    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
+        frequency_counter = Counter(nums)
+        unique_values = list(frequency_counter.keys())
+
+        def swap(list_object, index_a, index_b):
+            list_object[index_a], list_object[index_b] = list_object[index_b], list_object[index_a]
+
+        def partition(left, right, pivot_index):
+            pivot_frequency = frequency_counter[unique_values[pivot_index]]
+            # pivot を右端とswap(比較処理中に pivot が邪魔にならないように、一時的に末尾へ移動。)
+            swap(unique_values, right, pivot_index)
```

Comment:
unique_valuesへのアクセス方法について、swap関数内は引数経由でのローカルアクセス、partition関数内はグローバルアクセスになっているかと思いますが、不統一でやや違和感を持ちました。

自分なら、swapもローカルアクセスにしてしまうかなと思います。
swap関数も入れ替えたいインデックス2つのみを引数にしてしまえば、情報量が減って読みやすいかなって感じです。趣味の範囲かもしれません。

```suggestion
        def swap(index_a, index_b):
            unique_values[index_a], unique_values[index_b] = unique_values[index_b], unique_values[index_a]

        def partition(left, right, pivot_index):
            pivot_frequency = frequency_counter[unique_values[pivot_index]]
            # pivot を右端とswap(比較処理中に pivot が邪魔にならないように、一時的に末尾へ移動。)
            swap(right, pivot_index)
```

Reply:
ご指摘ありがとうございます、不統一感、気が付きませんでした。統一したほうがノイズにならずスッキリしますね。